### ISSUE TITLE
32 vs 64 bit in R

### ISSUE THREAD

**Issue by sthurner11** [FRAMverse/coding-practices#6]
I have ran into code from other folks which use packages which require using 32 bit R (mostly from Marlene). Just a very random thing to think about moving forward. Do you want best practices for this? If so, we may want to rewrite Marlene code.

**Comment by Ty-WDFW** [comment]
Was that code around connections to Access databases? That used to be a pain way back in the day, but I think has been largely resolved now via Windows updates.

**Comment by cbedwards-dfw** [comment]
We can add some language encouraging the use of packages/tools that are agnostic to 32 vs 64 bit, and/or leaving special comments identifying when this is likely to be an issue. I feel like I've occasionally encountered situations where the only R package that solves a problem is janky and ends up relying on specific architecture. But I think we can probably avoid those entirely.

**Comment by sthurner11** [comment]
I dug around quickly and I think you are right that it mostly relates to connecting to access databases (but in files from like 2021/2022). She seems to have 
" # Need to be sure to use 32 bit R (Tools, Global Options, Change, 32 bit, re-open Rstudio) "
In files where she uses the RODBC library to connect to access.

**Comment by Ty-WDFW** [comment]
Yup you used to have to use `RODBC::connectAccess()` for 32-bit R and `RODBC::connectAccess2007()` for 64-bit R. [The way RODBC connects to access](https://github.com/cran/RODBC/blob/a0e9f0aa2440156af7e3c3b8eb08eec62fe1dc6b/R/win.R#L74) (and `framrsquared` for that matter) is through the native windows driver which was updated 3-4 years ago, I would recommend just refactoring scripts to `RODBC::connectAccess2007()` from `RODBC::connectAccess()` and you shouldn't have to worry about 32-bit/64-bit anymore.

RODBC is a reaaallly old, the only benefits to using it over DBI is its ability to save data to an access database, if you don't need that DBI is probably better.

**Comment by cbedwards-dfw** [comment]
FWIW, `.Machine$sizeof.pointer` in R will return 8 for 64 bit R, and 4 for 32 bit R. So in general if it's necessary to make the scripts backwards compatible when using packages/tools that are different for 64 bit and 32 bit, you could include both versions of the code with an if statement based on the pointer size.
